command-line/lib/Command/Key/Public.hs

```diff
@@ -40,13 +43,25 @@ mod liftCmd = command "public" $
         <> progDesc "Get the public counterpart of a private key"
         <> footerDoc (Just $ string $ mconcat
             [ "The private key is read from stdin."
+            , "To get extended public key pass '--with-chain-code'."
+            , "To get public key pass '--without-chain-code'."
             ])
   where
     parser = Public
-        <$> encodingOpt [humanReadablePart|xpub|]
+        <$> encodingOpt [humanReadablePart|???|]
+        <*> publicOpt
 
 run :: Cmd -> IO ()
-run Public{encoding} = do
+run Public{encoding, chainCode} = do
     xprv <- hGetXPrv stdin
     let xpub = toXPub xprv
-    hPutBytes stdout (xpubToBytes xpub) encoding
+    (bytes, encoding') <- case chainCode of
```

Comment:
This line doesn't need to be an IO action. It could be a plain let or where clause.

I would be inclined to set up hrp in the CLI Parser, and use a function.

```haskell
let toBytes = case chainCode of
  WithChainCode -> xpubToBytes
  WithoutChainCode -> xpubPublicKey
hPutBytes stdout (toBytes xpub) encoding
```